File: banner/banner.py
#!/usr/bin/env python3
"""
yourTermux - Standalone banner script (Python)
Author: JubairSenseiDev

Uses ANSI escape codes for color (graceful fallback to plain text
when stdout is not a TTY or when NO_COLOR is set).

Usage:
    ./banner.py                  # default cyan banner
    python3 banner.py --color green
    python3 banner.py --no-color
    python3 banner.py --help
"""

from __future__ import annotations

import argparse
import os
import sys
from datetime import datetime

VERSION = "1.0.0"
BUILD_DATE = datetime.now().strftime("%d %B %Y")
AUTHOR = "JubairSenseiDev"
TAGLINE = "Termux setup for new users"

# ANSI color codes
ANSI = {
    "red":     "\033[1;31m",
    "green":   "\033[1;32m",
    "yellow":  "\033[1;33m",
    "blue":    "\033[1;34m",
    "magenta": "\033[1;35m",
    "cyan":    "\033[1;36m",
    "white":   "\033[1;37m",
    "dim":     "\033[2m",
    "reset":   "\033[0m",
}

BANNER_ART = [
    "    __   __ _______ _______ _______ ___   _______ ",
    "    \\ \\ / /|  ___  |  ___  |  ___  |   | |  _____|",
    "     \\ V / | |   | | |   | | |   | |   | | |_____ ",
    "      | |  | |   | | |   | | |   | |   | |_____  |",
    "      | |  | |___| | |___| | |___| |___| |_____| |",
    "      |_|  |_______|_______|_______|_______|_____|",
]


def colors_disabled(args) -> bool:
    if args.no_color:
        return True
    if os.environ.get("NO_COLOR"):
        return True
    if not sys.stdout.isatty():
        return True
    return False


def main() -> int:
    parser = argparse.ArgumentParser(
        prog="banner.py",
        description="yourTermux banner (Python)",
        formatter_class=argparse.RawDescriptionHelpFormatter,
    )
    parser.add_argument(
        "--color",
        choices=["red", "green", "yellow", "blue", "magenta", "cyan", "white"],
        default="cyan",
        help="Banner color (default: cyan)",
    )
    parser.add_argument(
        "--no-color",
        action="store_true",
        help="Disable ANSI colors",
    )
    args = parser.parse_args()

    if colors_disabled(args):
        c = y = w = g = m = b = d = r = ""
    else:
        c = ANSI[args.color]
        y = ANSI["yellow"]
        w = ANSI["white"]
        g = ANSI["green"]
        m = ANSI["magenta"]
        b = ANSI["blue"]
        d = ANSI["dim"]
        r = ANSI["reset"]

    # Banner art
    for line in BANNER_ART:
        print(f"{c}{line}{r}")

    # Info block
    print(f"    {y}Version {w}: {g}{VERSION}")
    print(f"    {y}Build   {w}: {g}{BUILD_DATE}")
    print(f"    {y}Author  {w}: {m}{AUTHOR}")
    print(f"    {y}Tagline {w}: {b}{TAGLINE}")
    print()
    print(f"    {d}Quick keys : F1=help  ESC=back  TAB=complete  CTRL+T=new session{r}")
    print(f"    {d}Commands   : chcolor  chfont  chzsh  pacupg  sd  pf{r}")
    print()

    return 0


if __name__ == "__main__":
    sys.exit(main())

File: banner/banner.js
#!/usr/bin/env node
/**
 * yourTermux - Standalone banner script (Node.js)
 * Author: JubairSenseiDev
 *
 * Pure Node.js — no npm dependencies required.
 * Uses ANSI escape codes for color (graceful fallback to plain text
 * when stdout is not a TTY or when NO_COLOR is set).
 *
 * Usage:
 *   ./banner.js                  # default cyan banner
 *   node banner.js --color green
 *   node banner.js --no-color
 *   node banner.js --help
 */

'use strict';

const VERSION  = '1.0.0';
const AUTHOR   = 'JubairSenseiDev';
const TAGLINE  = 'Termux setup for new users';

const BUILD_DATE = new Date().toLocaleDateString('en-US', {
  day:   '2-digit',
  month: 'long',
  year:  'numeric',
});

const ANSI = {
  red:     '\x1b[1;31m',
  green:   '\x1b[1;32m',
  yellow:  '\x1b[1;33m',
  blue:    '\x1b[1;34m',
  magenta: '\x1b[1;35m',
  cyan:    '\x1b[1;36m',
  white:   '\x1b[1;37m',
  dim:     '\x1b[2m',
  reset:   '\x1b[0m',
};

const BANNER_ART = [
  '    __   __ _______ _______ _______ ___   _______ ',
  '    \\ \\ / /|  ___  |  ___  |  ___  |   | |  _____|',
  '     \\ V / | |   | | |   | | |   | |   | | |_____ ',
  '      | |  | |   | | |   | | |   | |   | |_____  |',
  '      | |  | |___| | |___| | |___| |___| |_____| |',
  '      |_|  |_______|_______|_______|_______|_____|',
];

function parseArgs(argv) {
  const args = { color: 'cyan', noColor: false, help: false };
  for (let i = 0; i < argv.length; i++) {
    const a = argv[i];
    if (a === '--color') {
      args.color = argv[++i] || 'cyan';
    } else if (a === '--no-color') {
      args.noColor = true;
    } else if (a === '--help' || a === '-h') {
      args.help = true;
    } else if (a.startsWith('--color=')) {
      args.color = a.slice(8);
    }
  }
  return args;
}

function printHelp() {
  console.log(`yourTermux banner (Node.js)

Usage: node banner.js [OPTIONS]

Options:
  --color COLOR   Banner color: red, green, yellow, blue, magenta, cyan, white (default: cyan)
  --no-color      Disable ANSI colors
  --help, -h      Show this help message

Environment:
  NO_COLOR        If set, colors are disabled (respects https://no-color.org)`);
}

function main() {
  const args = parseArgs(process.argv.slice(2));

  if (args.help) {
    printHelp();
    process.exit(0);
  }

  const validColors = ['red', 'green', 'yellow', 'blue', 'magenta', 'cyan', 'white'];
  if (!validColors.includes(args.color)) {
    process.stderr.write(`Unknown color: ${args.color}\n`);
    process.stderr.write(`Valid colors: ${validColors.join(', ')}\n`);
    process.exit(2);
  }

  const disableColor = args.noColor
    || process.env.NO_COLOR
    || !process.stdout.isTTY;

  const c = disableColor ? '' : ANSI[args.color];
  const y = disableColor ? '' : ANSI.yellow;
  const w = disableColor ? '' : ANSI.white;
  const g = disableColor ? '' : ANSI.green;
  const m = disableColor ? '' : ANSI.magenta;
  const b = disableColor ? '' : ANSI.blue;
  const d = disableColor ? '' : ANSI.dim;
  const r = disableColor ? '' : ANSI.reset;

  for (const line of BANNER_ART) {
    console.log(`${c}${line}${r}`);
  }

  console.log(`    ${y}Version ${w}: ${g}${VERSION}`);
  console.log(`    ${y}Build   ${w}: ${g}${BUILD_DATE}`);
  console.log(`    ${y}Author  ${w}: ${m}${AUTHOR}`);
  console.log(`    ${y}Tagline ${w}: ${b}${TAGLINE}`);
  console.log('');
  console.log(`    ${d}Quick keys : F1=help  ESC=back  TAB=complete  CTRL+T=new session${r}`);
  console.log(`    ${d}Commands   : chcolor  chfont  chzsh  pacupg  sd  pf${r}`);
  console.log('');
}

main();
